pkg/client/kubeconfig_test.go

```diff
@@ -0,0 +1,173 @@
+phHGI9BG3jxGXsTRE2yAwE+8jdgnbBCM5Ro38J96pk+OmnYyWjgm8f76BnUG7hcL
+deI2t4uPx3i8HB76a+aYJr5FhNLFjkVELpxjZd87LTWEH3GAUOXROiBH30OJH9A9
+StST8iIcfS0CODYjxMg99IJaam8JqomSCgE8C4G9PJhgLiabVkvw91GxSuPMVZOd
+UrmxAqLW0KZr4p8Un11V5qxBUG3W1Rpr+lg9ZgzBb8NFm/Te5PgsTs+XZGhwX1TP
+IcLUdsNaYM20dfj3jO0+5K3qn/tCabm9y1k6vMJPwpC/eQ==
+-----END CERTIFICATE-----`
+
+	dummyKeyPem = `-----BEGIN RSA PRIVATE KEY-----
+MIIEowIBAAKCAQEAnevpBEsn6CWxT6DkD9RjJnqSAjducV+FAkWwxZAtGpwKJ2zV
+aD9x7zkKCpMmJi6kGD+sLZxhBu3hYEsJPzM0dEQbRPrChqhJbkSMm+ACJVIeb2Dh
+y30scMl3/JGcrEejK79iWfY77VYE5grTrS4RM7/VUGZEahlKKSHEWqLgxoNFeg8u
+3Im1+QwtJ4zYUNSTEIlyUmO1E6bvYaX7IibqEe2MszR4fV9BdzwisvWzXg+5t61D
+j0lXATgRGpkRul5gsglTjYNyWdJfY4enqlPPDiaF8WUggLnmzsBBUB3sPy4TuAI+
+yWa1riJRUIS39a/yzbsYODfGztsuvsFaoDlHQwIDAQABAoIBAF9AUK7XSf265mS6
+DXUCzL8DxRdzKblWPNqvAD1Zher73SAEg/+57NW2mLjiImt7TFyX4xkrrlZImtzC
+xZQKJYRPJAeKHFSuIoRQ8mJ+Ta0HB/Z0AB0Fpg1tZ2K+zToYh3G2oPLUEzdG3/OE
+6kIVfCizd01kMbWxBUsj49QrU9pHlqSZx6UNi8k+nGtpIjX9HARIZhCNqAeQyAwe
+HMPD8Jc569iAbDtp1dq2YELXWDssL7t5gqRa56dutxKFCIJvt2xDx8e0KQBeq+j4
+18ERVf8sf1DkuV9AdV5mQef1J8htexvjTUu36AFtYRwQuBWs8e9891WUqrag1Y8j
+gg73RgECgYEAweBMdX8WjdzfufV3l8I1Q3bnDI7DNifmP37nJJJRRfz1qWvXcPmz
+7dY4MgGdwOopJA3LfsyE3zCq7IEe+3LK6ZQV29QkheExXEGj9s3e1AHnpCEeqKOK
+qDMtxgOLpuh6hUZbSf069Fq7jpqo1c3lluIc9UStD2z3x+ApqDsJpkMCgYEA0IY+
+tDet1s73s/o9uzJ9LleN7ieLYSDxg5oKvZpn3A4c9Rrr7Fj1OBNOkPjlUsVRSb6f
+sMn1gz45cMc5eSSimen/R7MThcYUpN3EazWnZDS9DKN3PTwObONRj8g439AmIYLK
+y279PO3yxhg3B4BfXgkKPUe/BEXoRVDxg8n+SwECgYBwfb6fZjAl/ARsF7teeLcD
+ABirtqIZ6Ci2quFe3O7/VvkLZqFI0fnOhD9y9HEeID/ixYZPekeWYNysAXeCmmaW
+BPBx7rOKYtGLICMM7wLdrIVFPFpXqxym35sti50aKUX90obhdWchpQuygJZ6B8+x
+Ll1zCngHvUg/1xcUn7zHlQKBgG2F9S1HCWGH94Zqaz4FeMZ8aimqT4TGftO2dum7
+Tc3BA+ihKUVMPBAl4+A1Oo3M4bMwEkQS74btidH4cfF1Eopw4wpPvnNG5NTrPh1p
+YvA42wrmWNyqzJDYnKA+c9DqTPzQ658KPqxf9mGhmlWwUWbcrKofIu8loKe3qgKk
+d5IBAoGBALYKNPHZasv6D8z1HfubW3k/G3EzVXYrrNQ8mOnJjbGhuFh29/CGyodp
+LNSzVepEqDGARwkhMjpYMyXqUYlF+FpLo6WGmFxbhdTwD7uIAVf0vIfAhPzlaxeG
+9OT27Rx4qVWDEuy+wCdkYMfbnBbfreLCuPsHTkkl/kncbhpxGXIo
+-----END RSA PRIVATE KEY-----`
+)
+
+var _ = Describe("TLS Bootstrap kubeconfig tests", func() {
+	var (
+		mockCtrl            *gomock.Controller
+		imdsClient          *mocks.MockImdsClient
+		aadClient           *mocks.MockAadClient
+		tlsBootstrapClient  *tlsBootstrapClientImpl
+		mockReader          *mocks.MockfileReader
+		validKubeConfigPath string
+	)
+
+	Context("isKubeConfigStillValid Tests", func() {
+		BeforeEach(func() {
+			validKubeConfigPath = "dummykubeconfig"
+			_, err := os.Create(validKubeConfigPath)
+
+			if err != nil {
+				return
+			}
+
+			bootstrapClientConfig := &restclient.Config{
+				Host: "https://your-k8s-cluster.com",
+				TLSClientConfig: restclient.TLSClientConfig{
+					CertData: []byte(dummyCertPem),
+					KeyData:  []byte(dummyKeyPem),
+				},
+				BearerToken: "your-token",
+			}
+			err = writeKubeconfigFromBootstrapping(bootstrapClientConfig, validKubeConfigPath)
```

Comment:
we already have a filesystem interface we use everywhere which we can mock completely, I'd use that over trying to make temporarily file resources for testing

Reply:
So the problem is less that I want to make a file and more that everything wants a file path to our file. So its a lot easier to just make a temp file than to create a mockFileReader that has the byte contents of the file, but no "path"

Reply:
fair enough I suppose, though I'd still prefer avoiding creating temp file resources - we can keep as-is for now and maybe revisit after some further refactoring

I wouldn't necessarily be opposed to creating a dedicated type which is responsible for validation/generation of kubeconfig files - if we did that and a little tweaking, mocking file system calls would become much easier